change_analyzer/sequences_diff.py

```diff
@@ -170,21 +203,28 @@ def validate_actual_vs_expected_images(image_actual: str, image_expected: str) -
 
         return np.array_equal(image_expected_as_array, image_actual_as_array)
 
-    def save_step_images_from_arrays(self, image_actual: str, image_expected: str, filename_actual, filename_expected):
+    @staticmethod
+    def _convert_image_to_array(image: str) -> np.ndarray:
+        """Convert the given image from string to numpy ndarray"""
+        return np.array(json.loads(image), dtype=np.uint8)
 
-        folder = os.path.dirname(self.report_file)
+    @staticmethod
+    def _convert_array_image_to_pil(image: np.ndarray) -> PIL.Image.Image:
+        """Convert the given image from numpy ndarray to PIL image"""
+        return Image.fromarray(np.uint8(image))

```

Comment:
Probably oneliners could be used in the actual code without extracting it to separate functions.